src/services/websocket.rs

```diff
@@ -117,10 +193,16 @@ impl WebSocketTask {
 
 impl Task for WebSocketTask {
     fn is_active(&self) -> bool {
-        self.ws.ready_state() == SocketReadyState::Open
+        cfg_match! {
+            feature = "std_web" => self.ws.ready_state() == SocketReadyState::Open,
+            feature = "web_sys" => self.ws.ready_state() == WebSocket::OPEN,
+        }
     }
     fn cancel(&mut self) {
-        self.ws.close();
+        cfg_match! {
+            feature = "std_web" => self.ws.close(),
+            feature = "web_sys" => self.ws.close().unwrap(),
```

Comment:
We also need to drop the event listeners for web-sys here

Reply:
Yeah, how did I miss that 😮, done!